daiquiri/core/adapter/database/postgres.py

```diff
@@ -321,20 +321,28 @@ def _parse_column(self, row):
             'name': column_name
         }
 
-        if data_type.lower() in self.DATATYPES:
-            datatype = self.DATATYPES[data_type.lower()]
-        elif udt_name.lower() in self.DATATYPES:
+        if udt_name.lower() in self.DATATYPES: # universal types: int2, int4, int8, _float8, ...
             datatype = self.DATATYPES[udt_name.lower()]
+            
+        elif data_type.lower() in self.DATATYPES: # postgres types: smallint, bigint, ARRAY, ...
+            datatype = self.DATATYPES[data_type.lower()]
+            
         else:
             datatype = None
 
         if datatype:
-            column['datatype'] = datatype['datatype']
+            if '_' in udt_name.lower(): # ARRAY
+                column['datatype'] = datatype['datatype'] # the _ is still needed, will be removed by the serializer.
```

Comment:
But if I get it right, `datatype['datatype']` has no `_` here. Which is good since `datatype` is always assumed to be a IVOA datatype not a native postgres datatype.

Reply:
yes you are right, I forgot to remove the comment (rest from earlier not satisfying implementation). So indeed I use the udt_name (postgres) to figure out if the field is an array or not. So in principle we could do the same for MySQL with a `DataBaseAdapter.field_is_array()` method or something similar. The `datatype` itself is always IVOA compatible.